index.js

```diff
@@ -145,7 +148,7 @@ class HyperDHT extends DHT {
         signature: null
       }
 
-      ann.signature = Persistent.signAnnounce(target, token, from.id, ann, keyPair.secretKey)
+      ann.signature = signAnnounce(target, token, from.id, ann, keyPair.secretKey)
```

Comment:
async now? note the commit method can be async to help with that (it expects a promise returned)